Sts1CobcSw/Periphery/FramRingBuffer.hpp

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+
+#include <Sts1CobcSw/Periphery/Fram.hpp>
+#include <Sts1CobcSw/Serial/Serial.hpp>
+#include <Sts1CobcSw/Utility/Span.hpp>
+
+#include <rodos-debug.h>
+
+#include <cstddef>
+#include <cstdint>
+#include <span>
+
+
+namespace sts1cobcsw::fram
+{
+using sts1cobcsw::Span;
+using sts1cobcsw::TriviallySerializable;
+
+
+template<typename T, std::size_t poolSize>
+class RingBuffer
+{
+public:
+    // NOLINTBEGIN(*non-private-member-variables-in-classes)
+    std::uint64_t writeCnt = 0;
+    std::uint64_t readCnt = 0;
+    std::uint32_t occupiedCnt = 0;
+    Address startAddress;
+    // NOLINTEND(*non-private-member-variables-in-classes)
+
+    explicit RingBuffer(Address address) : startAddress(address)
+    {
+    }
+
+    RingBuffer() : startAddress(0){};
+
+    void Put(T const & newdata)
+    {
+        currentWrite_ = writeIndex_;
+        auto const address = startAddress + writeIndex_ * serialSize<T>;
+        fram::WriteTo(address, Span(Serialize(newdata)), 0);
+        writeIndex_++;
+
+        if(writeIndex_ >= poolSize)
+        {
+            writeIndex_ = 0;
+        }
+        if(occupiedCnt < poolSize)
+        {
+            occupiedCnt++;
+        }
+        writeCnt++;
+        currentWrite_ = UINT32_MAX;
+    }
+
+    // Get next item
+    void Get(T & fromRing)
+    {
+        // Jump the current being written record
+        if(readIndex_ == currentWrite_)
```

Comment:
This is never true since `currentWrite_` is always `UINT32_MAX` outside of `Put()`.